lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
I'm not sure what definitions of promise and future you're drawing from. Different languages and libraries use different definitions of "promise" and "future" ([Wikipedia's article for either is combined](https://en.wikipedia.org/wiki/Futures_and_promises)). The only constant is that they are similar, and even JDK8 allows chaining computation on an asynchronous result in addition to blocking to wait for that result.

Ultimately what I'm saying is the terms don't matter so much. The point is we need a way to represent a potentially asynchronous result, and queue up further computation when that result is available.

What constitutes "information on how to resolve itself"? In what way is blocking to wait for a future result "resolving itself?"